### ISSUE TITLE
Environment controls in git workflows to destroy aws pulumi stacks

### ISSUE THREAD

**Issue by shawngerrard** [asterion-digital/asterion-as-code#14]
To avoid accidental overuse of resources, we should introduce controls to destroy or keep pulumi stacks active, depending on environment or activity, when a [git workflow](https://github.com/asterion-digital/asterion-as-code/tree/master/.github/workflows) is triggered.

For instance, if not requiring a stack to be active, such as a stack produced in a dev environment, the stack may need to be destroyed upon workflow completion.

We can implement some type of conditional control within the git workflow to control this at a rudimentary level. We could also create a boolean environment secret to indicate to our workflow when to destroy stacks or when to keep them alive.

**Comment by jmhbnz** [comment]
Agreed during discussion on 20th April that by default our pipelines will teardown any deployed `infra - aws` and `app - <name>` stacks after creation.

The purpose of our aws workflows are simply to test that our stacks are deploying successfully so by default we should just throw away after a successful creation.

We can use an environment variable like `pulumi_destroy=true` as a way to toggle this though so we have the option of retaining created stacks when debugging etc.
